.taskcluster.yml

```diff
@@ -6,259 +6,212 @@ version: 1
-          scopes:
-            - queue:create-task:aws-provisioner-v1/github-worker
-            - queue:scheduler-id:${scheduler_id}
-            - secrets:get:project/mobile/android-components/public-tokens
-          payload:
-            maxRunTime: 3600
-            image: mozillamobile/android-components:1.11
-            env:
-              TASK_ID: ${decision_task_id}
-              SCHEDULER_ID: ${scheduler_id}
-              MOBILE_HEAD_REPOSITORY: ${repository}
-              MOBILE_HEAD_BRANCH: ${head_branch}
-              MOBILE_HEAD_REV: ${head_rev}
-            command:
-              - /bin/bash
-              - --login
-              - -cx
-              - >-
-                git fetch ${repository} ${head_branch}
-                && git config advice.detachedHead false
-                && git checkout ${head_rev}
-                && ./gradlew --no-daemon --version
-                && python automation/taskcluster/decision_task.py
-            features:
-              taskclusterProxy: true
-          metadata:
-            name: Android Components - Decision task
-            description: Schedules the build and test tasks for Android components.
-            owner: ${user}@users.noreply.github.com
-            source: ${repository}/raw/${head_rev}/.taskcluster.yml
-####################################################################################################
-# Task: Snapshot release
-####################################################################################################
-  - $if: 'tasks_for == "cron"'
-    then:
-      $let:
-        decision_task_id: {$eval: as_slugid("decision_task")}
-        expires_in: {$fromNow: '1 year'}
-        scheduler_id: focus-nightly-sched
-        user: ${event.sender.login}
-        repository: ${event.repository.clone_url}
-      in:
+  $let:
+    decision_task_id: {$eval: as_slugid("decision_task")}
+    expires_in: {$fromNow: '1 year'}
+    user: ${event.sender.login}
+
+    # We define the following variable at the very top, because they are used in the
+    # default definition
+    head_branch:
+      $if: 'tasks_for == "github-pull-request"'
+      then: ${event.pull_request.head.ref}
+      else:
+        $if: 'tasks_for == "github-push"'
+        then: ${event.ref}
+        else: ${event.release.target_commitish}
+
+    head_rev:
+      $if: 'tasks_for == "github-pull-request"'
+      then: ${event.pull_request.head.sha}
+      else:
+        $if: 'tasks_for == "github-push"'
+        then: ${event.after}
+        else: ${event.release.tag_name}
+
+    repository:
+      $if: 'tasks_for == "github-pull-request"'
+      then: ${event.pull_request.head.repo.html_url}
+      else: ${event.repository.html_url}
+
+    worker_type:
+      $if: 'tasks_for in ["github-push", "github-release"] && event.repository.html_url == "https://github.com/mozilla-mobile/android-components"'
+      then: mobile-3-decision
+      else: mobile-1-decision
```

Comment:
This implies that the decision-task's workerType for cron graphs is going to be `mobile-1-decision` if I get this right. Is this what we want?

Reply:
Oh nice catch! Fixed.